lib/ld-eventsource/impl/event_parser.rb

```diff
@@ -37,12 +37,16 @@ def items
               event = maybe_create_event
               reset_buffers
               gen.yield event if !event.nil?
-            else
-              case line
-                when /^(\w+): ?(.*)$/
-                  item = process_field($1, $2)
-                  gen.yield item if !item.nil?
-              end
+            elsif (colon = line.index(':'))
+              name = line.slice!(0...colon)
+              next unless name.match?(/^\w+$/)
```

Comment:
I don't think we need to do any regex matching here. The only decision we really need to make at this point is if there is any field name at all, or if this is a comment where the colon is the first character. If it's a comment, we can detect that simply by `colon` equaling 0 and then we know we can skip the line. In all other cases, there's no harm in just parsing out `name` and passing it to `process_field`, since `process_field` will ignore any field names it doesn't understand, regardless of whether they're made up of only `\w`-compliant characters or not.

Reply:
And since in normal usage an SSE server is only going to be sending valid field names like `event` and `data`, doing this match is imposing unnecessary overhead for the normal use case— that is, first it has to check that every character in `"event"` or `"data"` is in the `\w` charset, and _then_ we call `process_field` and it checks whether it is one of the known strings like `"event"` and `"data"`. If the server sends something like `(asdf#%-not-valid-chars: asjfkl;asdhfasdf`, then the regex match here is saving a little time because we know right away it's not a real field, but there's no reason to optimize for the "broken server" case rather than the "correct server" case.

Reply:
That's definitely true—I was trying to exactly duplicate the original functionality of the regex match without matching the entire string. I don't think the minor overhead of matching `asdf#%-not-valid-chars` against `\w` is that large, but we can still definitely omit the check if needed

Reply:
> I don't think the minor overhead of matching asdf#%-not-valid-chars against \w is that large

My point wasn't about having to match _invalid_ field names against `\w`, but rather, having to match all the _valid_ field names against `\w` too. That's serving absolutely no purpose as far as I can tell. I'm fairly sure that the only reason there was any mention of `\w` at all in the earlier code was that we had a regex for the whole line so we had to put _something_ on the left-hand side, and we chose to put `\w` rather than `[^:]`.

Reply:
In other words, I think what should happen in the normal case where it is a valid field name like `event` is that it computes `name` and `line`, and then it calls `process_field`, and the `when "event"` line in `process_field` matches that field name. And what should happen if we see an invalid field name like `asdf#%-not-valid` is that it computes `name` and `line`, and then it calls `process_field`, and there's no `when` for that value so it just falls through and does nothing. The regex match isn't really adding anything in either of those cases, except for the very small optimization of not having to compute `line` _in the invalid case_. But we want to optimize for the valid case, not the invalid one.